### PR TITLE
S14/cart page/feature/right side

### PR DESCRIPTION
**Pull request by vynnykpff**
**Check list**

- UI
- Add/Remove functionality
- Promo code functionality

**Promo codes:**

- XYZ-000 -> 5% discount
- XYZ-111 -> 8% discount
- XYZ-222 -> 10% discount
- REENBIT -> 30% discount

### CODE REVIEW

File: src/common/constants/ProductConstants/Product.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  description: {
+    short: "",
+    long: "",
+    full: [
+      {
+        title: "",
+        content: "",
+      },
+    ],
+  },
```
Comment (Yuliia143) [inline src/common/constants/ProductConstants/Product.ts]: Would be nice to split as another type - Description

File: src/common/constants/ProductConstants/Product.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  price: {
```
Comment (Yuliia143) [inline src/common/constants/ProductConstants/Product.ts]: Would be nice to split as another type - Price

File: src/common/constants/ProductConstants/Product.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    cost: 0,
+    regions: [""],
+  },
+  stock: {
+    amount: 0,
+    unitMeasure: "",
+  },
+  rating: 0,
+  images: [""],
+  reviews: [
```
Comment (Yuliia143) [inline src/common/constants/ProductConstants/Product.ts]: Would be nice to split as another type - Review

File: src/common/constants/ProductConstants/Product.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      content: "",
+      rating: 0,
+      userData: {
+        name: "",
+        image: "",
+      },
+      id: "",
+    },
+  ],
+  questions: [
```
Comment (Yuliia143) [inline src/common/constants/ProductConstants/Product.ts]: Would be nice to split as another type - Question

File: src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartPricingDetails/CartPricingDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+export const CartPricingDetails: FC<Props> = ({ cartProductsPayload, orderPromo }) => {
+  const pricingData: PricingItem[] = [
+    { title: "Subtotal", subtitle: `${getSubtotalPrice(cartProductsPayload)} USD` },
+    {
+      title: "Promo",
+      subtitle: `${orderPromo.discount}% ${getPriceWithPromo(getSubtotalPrice(cartProductsPayload), orderPromo.discount)} USD`,
+    },
+    { title: "Tax", subtitle: `17% ${getPriceWithTax(getSubtotalPrice(cartProductsPayload))} USD` },
```
Comment (Yuliia143) [inline src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartPricingDetails/CartPricingDetails.tsx]: Magic number

File: src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartTotalPriceAndDelivery/CartTotalPriceAndDelivery.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  orderPromo: CartPromocode;
+};
+
+export const CartTotalPriceAndDelivery: FC<Props> = ({ cartProductsPayload, cartProducts, orderPromo }) => {
+  const getTotalPrice = () => {
+    const subtotalPrice = getSubtotalPrice(cartProductsPayload);
+    const taxIncludedPrice = getPriceWithTax(subtotalPrice);
+    const priceWithPromo = getPriceWithPromo(subtotalPrice, orderPromo.discount);
+
+    return priceWithPromo ? subtotalPrice + taxIncludedPrice - priceWithPromo : subtotalPrice + taxIncludedPrice;
```
Comment (Yuliia143) [inline src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartTotalPriceAndDelivery/CartTotalPriceAndDelivery.tsx]: It is better to split into two variables subtotalPrice + taxIncludedPrice - priceWithPromo and subtotalPrice + taxIncludedPrice

File: src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartTotalPriceAndDelivery/CartTotalPriceAndDelivery.module.scss [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    color: var(--secondary-c2-color-500);
+    font-size: 26px;
+    font-weight: 600;
+    text-wrap: nowrap;
+
+    //.cartTotalPrice {
+    //  overflow: hidden;
+    //  text-overflow: ellipsis;
+    //}
+  }
```
Comment (Yuliia143) [inline src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartTotalPriceAndDelivery/CartTotalPriceAndDelivery.module.scss]: ?

**Review by Yuliia143 — DISMISSED** [review]

**Review by NikitaIASA — DISMISSED** [review]

**Pull request merged**